tags/tag_set.go

```diff
@@ -37,22 +36,24 @@ func (e ErrKeyNotFound) Error() string {
 	return fmt.Sprintf("key %q not found", e.Key)
 }
 
-// TagSet contains a set of tags. Use TagSetBuilder to build tag sets.
+// TagSet contains a set of tags. Use NewTagSet to build tag sets.
 type TagSet struct {
 	m map[Key][]byte
```

Comment:
Is `TagSet` intended to be immutable?  If so, exporting Mutator seems to work in opposition to that.  If not, then this type seems simpler to describe as just `type TagSet map[Key][]byte`, much like `url.Values`.

Reply:
We want users not to care about encoding to []byte. Mutators are working against the types they know. UpdateInt64 will take a string and update the tag set by encoding the int64 into []byte. I'd prefer to hide the low-level representation from the user as much as possible.

Reply:
Ah, so if you are trying to abstract the marshaling, what about this:

```go
type Set map[Key]Value  // since package is tags, this is tags.Set

func (set Set) Copy() Set

type Value struct {
  b []byte
}

func StringValue(s string) Value {
  return Value{[]byte(s)}
}

func Int64Value(i int64) Value {
  return Value{[]byte{/* encode the int */}}
}
```

Seems like it would be simpler to provide functions to copy the map and write the modifications in plain Go than creating an operation per operation x data type pair.  WDYT?